File: templates/java/v1/singleton/AppConfig.java
package singleton;

import java.util.concurrent.ConcurrentHashMap;
import java.util.concurrent.ConcurrentMap;

public final class AppConfig {
    private final ConcurrentMap<String, String> values = new ConcurrentHashMap<>();

    private AppConfig() {}

    private static class Holder {
        private static final AppConfig INSTANCE = new AppConfig();
    }

    public static AppConfig getInstance() {
        return Holder.INSTANCE;
    }

    public void set(String key, String value) {
        values.put(key, value);
    }

    public String get(String key, String defaultValue) {
        return values.getOrDefault(key, defaultValue);
    }
}

File: templates/javascript/v1/singleton/appConfig.js
const configValues = new Map();
const EMPTY_DEFAULT = "";

const appConfig = {
  set(key, value) {
    configValues.set(key, value);
  },
  get(key, defaultValue = EMPTY_DEFAULT) {
    return configValues.has(key) ? configValues.get(key) : defaultValue;
  }
};

Object.freeze(appConfig);

export { appConfig };
